Frameworks/Foundation/NSOperation.mm

```diff
@@ -119,10 +125,10 @@ - (BOOL)isReady {
  @Status Interoperable
 */
 - (void)addDependency:(id)operation {
-    std::lock_guard<std::recursive_mutex> lock(_dependenciesLock);
+    std::lock_guard<std::recursive_mutex> dependenciesLock(_dependenciesLock);
     [self willChangeValueForKey:@"dependencies"];
     [_dependencies addObject:operation];
```

Comment:
NSKeyValueObservingOptionInitial will cause observeValueForKeyPath to be invoked on the dependency, which contains the logic you mentioned.

---
In reply to: [72369304](https://github.com/Microsoft/WinObjC/pull/691#discussion_r72369304) [](ancestors = 72369304)